### PR TITLE
ENG-18476: Ability to configure number of retention policy threads.

### PR DESCRIPTION
**Pull request by manjujames**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 1]
```diff
@@ -57,6 +57,20 @@
  * @param M Type of extra header metadata stored in the PBD
  */
 public class PersistentBinaryDeque<M> implements BinaryDeque<M> {
+    private static RetentionPolicyMgr s_retentionPolicyMgr;
```
Comment (n3world) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java:60]: How does this work with system tests which start and stop multiple local volt instances? Can this be shutdown and recreated?

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 2]
```diff
@@ -57,6 +57,20 @@
...
+
+    public static synchronized void setupRetentionPolicyMgr(int numThreads) {
+        if (s_retentionPolicyMgr == null) {
+            s_retentionPolicyMgr = new RetentionPolicyMgr(numThreads);
+        } else if (s_retentionPolicyMgr.getRetentionThreadPoolSize() != numThreads) {
+            s_retentionPolicyMgr.updateThreadPoolSize(numThreads);
```
Comment (n3world) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java:66]: Why not just always call the update and have the update either apply the change if it is different or ignore it

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java (hunk 1 above)
Comment (manjujames) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java:60]: I am not sure I understand the issue.

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java (hunk 2 above)
Comment (manjujames) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java:66]: Sure, that will work too.

File: tests/frontend/org/voltdb/utils/TestTimeBasedRetentionPolicy.java [hunk 3]
```diff
@@ -270,6 +271,11 @@ private void writeBuffers(int numBuffers) throws Exception {
         }
     }
 
+    @BeforeClass
+    public static void setUpClass() {
+        PersistentBinaryDeque.setupRetentionPolicyMgr(2);
```
Comment (rdykiel) [inline tests/frontend/org/voltdb/utils/TestTimeBasedRetentionPolicy.java:276]: You might want a test that verifies that the time-based retention continues to work properly after a change in the number of threads

Comment (manjujames) [reply tests/frontend/org/voltdb/utils/TestTimeBasedRetentionPolicy.java:276]: I will modify a couple of tests to change the thread count for both time and max bytes policies.

**Comment by manjujames** [comment]
Going to create a new PR against master.

**Pull request closed without merging**